cpukit/include/rtems/score/stackprotection.h

```diff
@@ -0,0 +1,177 @@
+ * notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright 
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+ * POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+
+#ifndef _RTEMS_SCORE_STACKPROTECTION_H
+#define _RTEMS_SCORE_STACKPROTECTION_H
+
+#if defined ( ASM )
+  #include <rtems/asm.h>
+#else
+  #include <rtems/score/basedefs.h>
+  #include <rtems/score/memorymanagement.h>
+  #include <rtems/score/chainimpl.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined ( ASM )
+
+/**
+ * The following defines the attributes of a protected stack
+ */
+typedef struct
+{  
+  /** This is the stack address */
+  uintptr_t        stack_address;
+  /** This is the stack size */
+  size_t        size;
+  /** This is the pointer to the page table base */
+  uintptr_t      page_table_base;
+  /**Memory flag for the alllocated/shared stack */
+  Memorymanagement_flags  access_flags;
+} Stackprotection_Attr;
+
+/**
+ * The following defines the control block  of a shared stack
+ */
+typedef struct
+{
+  /** This is the chain node for tracking shared stacks */
+  Chain_Node node;
+  /** This is the attribute of a shared stack*/
+  Stackprotection_Attr    Base;
+} Stackprotection_Shared_stack;
+
+/**
+ * The following defines the control block of an allocated stack 
+ */
+typedef struct
+{ 
+  /** This is the attribute of an allocated stack*/
+  Stackprotection_Attr    Base;
+  /** This is the pointer to the attributes of a stack shared with the stack 
+   * in question
+   */
+  Stackprotection_Shared_stack  *shared_stacks; 
+  /** 
+  * The chain control for tracking the shared stacks with the thread-stack in 
+  * question.
+  */
+  Chain_Control shared_node_control;
+} Stackprotection_Stack;
+
+/**
+ * The following defines the control block for handling names of stack-address 
+ */
+
+typedef struct 
+{
+  /** The address of the allocated stack */
+  uintptr_t stack_address;
+  /** Name of the allocated stack */
+  char *name;
+}Stackprotection_Stack_name;
+
+/**
+ * @brief Share a stack with another stack.
```

Comment:
you don't share a stack with a stack, you share it with a thread? This is kind of confusing to me.